.github/workflows/semgrep.yml

```
@@ -16,7 +16,7 @@ jobs:
     env:
       SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
     container:
-      image: returntocorp/semgrep
+      image: returntocorp/semgrep@sha256:0f12ff85e0e3d7e586190196c723e2fc47c4f3d99c760c37a74c1ee1fffb7bff
     steps:
-    - uses: actions/checkout@v4
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
```

```suggestion
    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Refs: https://github.com/OpenINF/.github/pull/646/files